api/v1beta1/module_types.go

```diff
@@ -79,6 +96,10 @@ type Build struct {
 	// +optional
 	// KanikoParams is used to customize the building process of the image.
 	KanikoParams *KanikoParams `json:"kanikoParams,omitempty"`
+
+	// +optional
+	// Sign enables in-cluster signing for this mapping
+	Sign *Sign `json:"sign"`
```

Comment:
Why is Sign under the Build? What if we want to sign a pre-built image?

Reply:
because the code (i.e. this PR) sits under the build code, so it feels to me that is should do in the CRD as well. I see the "build" process as "create an image we can use" not just "create an image", and if the kmod isn't signed we cant use it.

 It also means I can leverage helper.GetRelevantBuild() to calculate the config to use from the kernelmapping which makes life simpler.  

If we're signing a pre-built image you just need kernelmapping.build.sign e.g.
```
      kernelMappings:
        - regexp: '^.*\.x86_64$'
          containerImage: quay.io/chrisp262/minimal-driver:test1-signed6
          build:
            sign:
              unsignedImage: quay.io/chrisp262/minimal-driver:procfsv1
              keySecret:
                name: securebootkey
              certSecret:
                name: securebootcert
              filesToSign:
                - /modules/simple-kmod.ko
                - /modules/simple-procfs-kmod.ko
```
I am open to moving it to be a peer of build in the crd (so kernelmapping.build and kernelmapping.sign) if people prefer.  (I originally had it this way, with sign being its own package alongside job but it turned into a maintainability nightmare)

Reply:
i personally think that it is confusing: build is associated with in-cluster build, and signing is a little different. But lets have other people's opinion: @qbarrand @ybettan

Reply:
The (kaniko) build and signing are very similar tasks, most of the differences are in the job pod they run so separating off signing completely leads to a lot of duplicated job-management code, basically a copy/paste of internal/build/job/manager with the field names changed, and that feels like a maintenance mess of its own verses making the build manager a generic job manager and using the same code for both.

I did try using the reconciler to run the jobs (basically running handleBuild() twice, for build then sign), which is in some ways a nicer design, but it involves more and wider changes in the code which is still changing a lot and meant I was spending more time in merge hell trying to rebase onto main and keep up with changes from other PRs then on making this feature work (one step forward two steps back!). So this design is an attempt to balance between the scope of changes it makes and code reuse.

On a narrower point I'm not sure I see signing as separate from building, its a step in the build process. If you're using secure boot then the kaniko output is just an intermediate artifact that isn't usable until the kmods are signed. So the steps are separate in the same sense creating object files and linking are separate processes, technically they are, but in reality they both need to be done to build something useful.

Reply:
the issue is not the kaniko output, but that customer can use the signing option without kaniko at all: he has pre-built images, but instead of trying to sign them off-cluster ( which can be painful job), he will just lets us sign them in-cluster. In which case it does not make sense for customer to define Build context, when no actual build is being done. Regarding code split: how is it going to be done in downstream?  There we have build config for image building, but we won't use it for signing, correct?

Reply:
but whether the intermediate image comes from kaniko or from outside you are still building it for use in the cluster, so I think it does make sense to define a Build context.  I guess the argument is about whether Build is "create a usable image" or "create an intermediate image we need to do more stuff to before kmmo can use it", I'm (clearly :) ) going for the first option.

As for buildconfig, iirc that code still implements the Maker interface so we can still replace the maker instance with it (or with some cleverness in ShouldRun() have kaniko, buildconfig, and signing in the same binary and check which ones to use at runtime)